RecoTracker/DisplacedRegionalTracking/plugins/DisplacedRegionSeedingVertexProducer.cc

```diff
@@ -133,11 +156,13 @@ void DisplacedRegionSeedingVertexProducer::produce(edm::StreamID streamID,
     const auto distancePred = [](const Distance &a) {
       return (!a.entities().first->valid() || !a.entities().second->valid());
     };
-    const auto pseudoROIPred = [](const DisplacedVertexCluster &a) { return !a.valid(); };
-    distances.remove_if(distancePred);
-    pseudoROIs.remove_if(pseudoROIPred);
+    itLast = std::remove_if(itBegin, itLast, distancePred);
   }
 
+  const auto pseudoROIPred = [](const DisplacedVertexCluster &a) { return !a.valid(); };
+  auto remove_it = std::remove_if(pseudoROIs.begin(), pseudoROIs.end(), pseudoROIPred);
+  pseudoROIs.erase(remove_it, pseudoROIs.end());
```

Comment:
You can possibly save some time by also removing this line, then in the next call to `std::remove_if` do
```
auto remove_it2 = std::remove_if(pseudoROI.begin(), remove_it, roiPred);
```
Then the final `pseudoROIs.erase(...)` can can be removed if the last loop over the `pseudoROIs` is
```
for (auto it = pseudoROI.begin(); it != remove_it2; ++it) {
        auto const& roi = *it;
```